be/src/exec/spill/spill_components.h

```diff
@@ -178,15 +173,27 @@ class RawSpillerWriter final : public SpillerWriter {
 
     Status yieldable_flush_task(workgroup::YieldContext& ctx, RuntimeState* state, const MemTablePtr& mem_table);
 
+    void add_block_group(BlockGroupPtr&& block_group) { _block_group_set.add_block_group(std::move(block_group)); }
+
 public:
     struct FlushContext : public SpillIOTaskContext {
         std::shared_ptr<SpillOutputDataStream> output;
+        std::shared_ptr<BlockGroup> block_group;
+        InputStreamPtr input_stream;
+        // only used in DCHECK
+        size_t compact_input_num_rows{};
```

Comment:
ifdef ?

Reply:
In some cases release mode also requires this CHECK to be turned on.